ivi-layermanagement-examples/EGLWLMockNavigation/src/ShaderTexture.cpp

```diff
@@ -0,0 +1,91 @@
+/***************************************************************************
+ *
+ * Copyright 2010,2011 BMW Car IT GmbH
+ *
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ ****************************************************************************/
+#include "ShaderTexture.h"
+
+const static char* vertexShaderCode =
+        "attribute mediump vec4 a_vertex;                                \
+         uniform mediump mat4 u_modelMatrix;                             \
+         attribute mediump vec2 a_texCoord;                              \
+         varying mediump vec2 v_texCoord;                                \
+         void main(void)                                                 \
+         {                                                               \
+            gl_Position = u_projectionMatrix * u_modelMatrix * a_vertex; \
+            v_texCoord = a_texCoord;                                     \
+         }";
+
+const static char* fragmentShaderCode =
+        "varying mediump vec2 v_texCoord;                    \
+         uniform sampler2D s_texture;                        \
+         void main (void)                                    \
+         {                                                   \
+            gl_FragColor = texture2D(s_texture, v_texCoord); \
+         }";
+
+ShaderTexture::ShaderTexture(float* projectionMatrix)
+: ShaderBase(vertexShaderCode, fragmentShaderCode, projectionMatrix)
+{
+    glUseProgram(shaderProgramId);
+    m_uniformModelMatrix = glGetUniformLocation(shaderProgramId, "u_modelMatrix");
+    m_uniformSampler = glGetUniformLocation(shaderProgramId, "s_texture");
+    attributeTexCoord = glGetAttribLocation(shaderProgramId, "a_texCoord");
+}
+
+ShaderTexture::~ShaderTexture()
+{
```

Comment:
don't you need to delete the shader here ?

Reply:
is it done in base class ?

Reply:
Yes, it is done in ShaderBase destructor, where ShaderBase::destroyShader() is called.